weighted_levenshtein/clev.pyx

```diff
@@ -228,8 +226,11 @@ cdef DTYPE_t c_damerau_levenshtein(
         unsigned int char_i, char_j
         DTYPE_t cost, ret_val
         Py_ssize_t db, k, l
-
         Array2D d
+        Py_ssize_t len1, len2
+
+    len1 = str1.shape[0]
```

Comment:
Just to make sure. does `str1.size` not work here? I noticed you used `.size` before.

Reply:
This is in the Cython documentation:
_"arr_memview.shape[0] could have been replaced by arr_memview.size, arr.shape[0] or arr.size. But arr_memview.shape[0] is more efficient because it doesn’t require any Python interaction."_ - [Source](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#pass-data-from-a-c-function-via-pointer)
So that's why I used this.

Reply:
Cool, makes sense. Thanks for the link!